### PR TITLE
Add `SMISMEMBER` command.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
https://redis.io/commands/smismember/

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 1]
```diff
@@ -75,4 +75,20 @@ public interface SetBaseCommands {
...
      */
     CompletableFuture<Long> scard(String key);
+
+    /**
+     * Returns whether each <code>member</code> is a member of the set stored at <code>key</code>.
+     *
+     * @see <a href="https://redis.io/commands/smismember/">redis.io</a> for details.
+     * @param key The key of the set to check.
+     * @param members A list of members to check their presence.
+     * @return An array of boolean values indicating the members presence. Each array element reflects
```
[the remark this answers is not in the document]
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/commands/SetBaseCommands.java:85]: An array of boolean values, each indicating if the respective member exists in the set.

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 2]
```diff
@@ -824,6 +824,27 @@ public void sadd_srem_scard_smembers_key_with_non_set_value(BaseClient client) {
...
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void smismember(BaseClient client) {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+
+        assertEquals(2, client.sadd(key1, new String[] {"1", "2"}).get());
+        assertArrayEquals(
+                new Boolean[] {true, false}, client.smismembmer(key1, new String[] {"1", "3"}).get());
+        // empty set
```
Comment (SanHalacogluImproving) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:837]: Not empty set the key does not exist in this example

**Comment by SanHalacogluImproving** [comment]
Not super knowledged on the Rust side but left comments for the java side.

File: glide-core/src/client/value_conversion.rs [hunk 3]
```diff
@@ -159,6 +160,21 @@ pub(crate) fn convert_to_expected_type(
...
+            Value::Array(array) => {
+                let array_of_bools = array
+                    .iter()
+                    .map(|v| Value::Boolean(from_owned_redis_value::<bool>(v.clone()).unwrap()))
+                    .collect();
+                Ok(Value::Array(array_of_bools))
+            }
+            _ => Err((
+                ErrorKind::TypeError,
+                "Response couldn't be converted to an array of {boolean}",
```
Comment (jonathanl-bq) [inline glide-core/src/client/value_conversion.rs:173]: Why do you have braces around "boolean"? That's not the array syntax in Rust, and also reads like "an array of array of boolean"

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 4]
```diff
@@ -75,4 +75,20 @@ public interface SetBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> scard(String key);
+
+    /**
+     * Returns whether each <code>member</code> is a member of the set stored at <code>key</code>.
+     *
+     * @see <a href="https://redis.io/commands/smismember/">redis.io</a> for details.
+     * @param key The key of the set to check.
+     * @param members A list of members to check their presence.
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/SetBaseCommands.java:84]: We already have list in other list commands. Do you want to do a PR to update them all?

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 2 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:837]: empty set = key does not exist
You can't create an empty set; if you remove all elements from it - key would be deleted

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 5]
```diff
@@ -75,4 +75,20 @@ public interface SetBaseCommands {
...
+     * @return An array of boolean values indicating the members presence. Each array element reflects
+     *     presence of the corresponding member in <code>members</code> argument.
+     * @example
+     *     <pre>{@code
+     * Boolean[] presence = client.smismembmer("my_set", new String[] { "a", "b", "c" }).get();
+     * assert presence[0] && presence[1] && !presence[2]; // Only first two elements are present in the set
+     * }</pre>
+     */
+    CompletableFuture<Boolean[]> smismembmer(String key, String[] members);
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/SetBaseCommands.java]: 🤦

**Review by jonathanl-bq — APPROVED** [review]
Signing off on the Rust portion only.

File: glide-core/src/client/value_conversion.rs [hunk 6]
```diff
@@ -221,6 +237,7 @@ pub(crate) fn expected_type_for_cmd(cmd: &Cmd) -> Option<ExpectedReturnType> {
         b"HEXISTS" | b"HSETNX" | b"EXPIRE" | b"EXPIREAT" | b"PEXPIRE" | b"PEXPIREAT"
         | b"SISMEMBER" | b"PERSIST" => Some(ExpectedReturnType::Boolean),
         b"SMEMBERS" => Some(ExpectedReturnType::Set),
+        b"SMISMEMBER" => Some(ExpectedReturnType::ArrayOfBools),
```
Comment (acarbonetto) [inline glide-core/src/client/value_conversion.rs:240]: Is Bool and Boolean the same in Rust?

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 7]
```diff
@@ -75,4 +75,19 @@ public interface SetBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> scard(String key);
+
+    /**
+     * Returns whether each member is a member of the set stored at <code>key</code>.
+     *
+     * @see <a href="https://redis.io/commands/smismember/">redis.io</a> for details.
+     * @param key The key of the set to check.
+     * @param members A list of members to check their presence.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:84]: Presence? This is from sismember: 
```suggestion
     * @param members A list of members to check for existence in the set.
```

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 8]
```diff
@@ -75,4 +75,19 @@ public interface SetBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> scard(String key);
+
+    /**
+     * Returns whether each member is a member of the set stored at <code>key</code>.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:80]: I know this is from redis.io, but consider: 
```suggestion
     * Checks whether each member is contained in the members of the set stored at <code>key</code>.
```

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 9]
```diff
@@ -75,4 +75,19 @@ public interface SetBaseCommands {
...
+     *
+     * @see <a href="https://redis.io/commands/smismember/">redis.io</a> for details.
+     * @param key The key of the set to check.
+     * @param members A list of members to check their presence.
+     * @return An array of boolean values, each indicating if the respective member exists in the set.
+     * @example
+     *     <pre>{@code
+     * Boolean[] presence = client.smismembmer("my_set", new String[] { "a", "b", "c" }).get();
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:88]:
```suggestion
     * Boolean[] isMembers = client.smismember("my_set", new String[] { "a", "b", "c" }).get();
```

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 10]
```diff
@@ -75,4 +75,19 @@ public interface SetBaseCommands {
...
+     * assert presence[0] && presence[1] && !presence[2]; // Only first two elements are present in the set
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:89]:
```suggestion
     * assert isMembers[0] && isMembers[1] && !isMembers[2]; // Only first two elements are members of the set
```

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 11]
```diff
@@ -824,6 +824,27 @@ public void sadd_srem_scard_smembers_key_with_non_set_value(BaseClient client) {
...
     }
 
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void smismember(BaseClient client) {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+
+        assertEquals(2, client.sadd(key1, new String[] {"1", "2"}).get());
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:834]:
```suggestion
        assertEquals(2, client.sadd(key1, new String[] {"one", "two"}).get());
```

**Review by acarbonetto — COMMENTED** [review]
Minor comments

File: glide-core/src/client/value_conversion.rs (hunk 6 above)
Comment (Yury-Fridlyand) [reply glide-core/src/client/value_conversion.rs:240]: image

Comment (Yury-Fridlyand) [reply glide-core/src/client/value_conversion.rs:240]: rust has bool, not boolean

**Pull request merged**